### ISSUE TITLE
[Feature Request] ElOption   resetIndex $emit   crash

### ISSUE THREAD

**Issue by Warlock-developer** [ElemeFE/element#5790]
### Existing Component
Yes

### Component Name
el-select

### Description
Hello, I have the next component.

<label for="department" class="col-sm-12 control-label">Departamento<small class="text-danger">*</small></label>
            <el-select v-model="department" filterable clearable placeholder="Seleccionar Departamento"
                       name="department" class="col-lg-12" >
                <el-option
                        v-for="department in listDepartments"
                        :label="department.label"
                        :key="department.value"
                        :value="department.value">
                </el-option>
            </el-select>

this component belongs to a form, and i have a close button for the form, When I push the close buttom, vuedevtools register 1600 events generated by ElOption resetIndex $emit by ElOption. 

export default{
        props: props,

        name: 'select-department',

        components: {
            'el-select': Select,
            'el-option': Option,
            'form-error': FormError
        },

        data(){
            return {
                department: null,
                listDepartments: [],
            }
        },
mounted(){
            this.getDepartments() /// this method get and array of departments.
        },

I am using <component :is="showView"></component> in the principal form to show and hide the form.

Thanks for the help

**Comment by Leopoldthecoder** [comment]
Issues without a reproduction link will be closed directly.

**Comment by Warlock-developer** [comment]
you can use your https://jsfiddle.net/api/post/library/pure/, if you can run vuedevtools, set the options variable to options = []. the vuedevtools register 20 resetIndex emit by eloption, and I have a componen with 30 options, this show 1200 resetindex events

**Comment by yunhuduan** [comment]
我也碰到这个问题了，我使用el-option改变data后600多个resetindex；我使用原生的select就没有这个问题。

**Comment by lerit** [comment]
是的，不打开vue-devtools感觉不到，一开发，发现里面的events数量猛增，卡顿严重，都是resetindex.
![image](https://user-images.githubusercontent.com/4743467/28526760-108ff1fe-70bb-11e7-828d-d6ec84faede8.png)

**Comment by wacky6** [comment]
@Warlock-developer Sorry, I totally missed this issue. Thanks for discovering this problem. I have experienced this several times, but never bothered to dig deeper since it causes no problem in production.

@Leopoldthecoder I strongly suggest reopening this issue.

---

This is a valid performance issue, which severely impacts developer experience. It almost hangs vue-dev-tools (if open) and chrome.

Have hacked an 'easy-to-view' demo, watch for repeat counter in console. https://jsfiddle.net/fnx7qh06/

---

**Without dev-tools:**
![image](https://user-images.githubusercontent.com/9319556/28529758-b4806aaa-70d4-11e7-9c47-e7ba14f99129.png)

**With dev-tools:**
![image](https://user-images.githubusercontent.com/9319556/28530138-180cce82-70d6-11e7-919d-5a8376f73c77.png)

The impact is so severe that I only profiled for a short time. The total time it takes for CPU to go down from 100% to idle takes about 2-3 minutes for 100 options.

**Comment by wacky6** [comment]
The problem seems to be the devtool trying to "serialize" the `onOptionDestroy` event parameter.

The parameter is a vue instance, which is extremely costly to be serialized. Changed the argument to `() => this`, then everything is fast again.

I think this issue mostly is devtool's problem. Though I would like to see some simplification on Select's implementation. The current broadcast / dispatch pattern is somewhat discouraged by vue.

@Leopoldthecoder

**Comment by Leopoldthecoder** [comment]
@wacky6 That's why I'm planning to manage all data inside Select itself by removing `el-option` entirely and replacing it with a `options` prop on `el-select`.

**Comment by wacky6** [comment]
`options` prop is going to be a HUGE breaking change. I prefer less disruptive solutions (if can be found, ant.d maybe?)

**Comment by Leopoldthecoder** [comment]
Turns out I don't have to remove `el-option`. Instead, I removed `resetIndex`.

Previously, when an option is destroyed, every option emits a `resetIndex` event. So in the demo https://jsfiddle.net/fnx7qh06/ , when you destroy 100 options, `resetIndex` is triggered 5050 (1 + 2 + 3 + ... + 100) times.

`resetIndex` is removed in #6151. `onOptionDestroy` still triggers each time an option is destroyed. Good news is, the parameter of `onOptionDestroy` is changed from a Vue instance to a number. Now destroying 100 options with dev-tool:
![image](https://user-images.githubusercontent.com/10095631/28745824-eb325ab8-74b2-11e7-8f1e-120c3a8dc7ff.png)
